cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-logging.adoc

```diff
@@ -0,0 +1,219 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="cloud-experts-deploying-application-logging"]
+= Tutorial: Logging
+include::_attributes/common-attributes.adoc[]
+include::_attributes/attributes-openshift-dedicated.adoc[]
+:context: cloud-experts-deploying-application-logging
+
+toc::[]
+
+//rosaworkshop.io content metadata
+//Brought into ROSA product docs 2024-05-01
+
+There are various methods to view your logs in {product-rosa} (ROSA). Use the following procedures to forward the logs to AWS CloudWatch, view the logs directly through the pod by using `oc logs`, and use Kibana (EFK Stack) to search the logs.
+
+[NOTE]
+====
+ROSA is not preconfigured with a logging solution.
+====
+
+The cluster logging components are based on Fluentd, Elasticsearch (ES), and Kibana. Fluentd is deployed to each node in the cluster. It collects application logs and either writes them to ES or forwards them to CloudWatch. Kibana is a centralized, web-based user interface (UI) where users and administrators can create visualizations and dashboards with the aggregated data.
+
+.Prerequisites
+
+* Set up a logging solution before viewing your logs. See the xref:../../observability/logging/cluster-logging-deploying.adoc#cluster-logging-deploying[Installing logging] documentation for more information.
+
+.Additional resources
+* xref:../../observability/logging/cluster-logging.adoc#cluster-logging[Cluster logging]
+
+== Installing the logging add-on service
+
+Install the logging add-on service to forward the logs to AWS CloudWatch.
+
+//These steps were adopted from the Managed OpenShift Black Belts xref:../../cloud_experts_tutorials/cloud-experts-rosa-cloudwatch-sts.adoc#cloud-experts-rosa-cloudwatch-sts[tutorial].
+
+. Run the following script to configure your ROSA cluster to forward logs to CloudWatch:
++
+[NOTE]
+====
+Configuring ROSA to send logs to CloudWatch goes beyond the scope of this tutorial. Integrating with AWS and enabling CloudWatch logging are important aspects of ROSA, so a script has been included to simplify the configuration process. The script will automatically set up AWS CloudWatch. You can examine the script to understand the steps involved.
+====
++
+[source,terminal]
+----
+$ curl https://raw.githubusercontent.com/openshift-cs/rosaworkshop/master/rosa-workshop/ostoy/resources/configure-cloudwatch.sh | bash
```

Comment:
OperatorGroup is not configured in openshift-logging namespace. how about add a step to create operatorgroup?

Reply:
https://docs.openshift.com/container-platform/4.15/observability/logging/cluster-logging-deploying.html#cluster-logging-deploy-cli_cluster-logging-deploying

Reply:
spec.collection.logs.* is deprecated in favor of spec.collection.*. Can you update the shell?
```
apiVersion: logging.openshift.io/v1
kind: ClusterLogging
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"logging.openshift.io/v1","kind":"ClusterLogging","metadata":{"annotations":{},"name":"instance","namespace":"openshift-logging"},"spec":{"collection":{"logs":{"type":"fluentd"}},"forwarder":{"fluentd":{}},"managementState":"Managed"}}
  creationTimestamp: "2024-06-26T02:18:57Z"
  generation: 1
  name: instance
  namespace: openshift-logging
  resourceVersion: "78015"
  uid: 9cac7c44-70c9-4030-8e36-1cf5ec3f4873
spec:
  collection:
    logs:
      type: fluentd
  forwarder:
    fluentd: {}
  managementState: Managed
status:
  conditions:
  - lastTransitionTime: "2024-06-26T02:18:57Z"
    message: spec.collection.logs.* is deprecated in favor of spec.collection.*
    reason: DeprecatedCollectionLogsSpec
    status: "True"
    type: Deprecated
  logStore: {}
  visualization: {}

```

Reply:
updated configure-cloudwatch.sh  https://privatebin.corp.redhat.com/?1a9ed31234401ace#JBiu5Avp3mNdQSCmVBpnExwBFL2si38RV5DWktG9XQbZ

Reply:
Please ask the script owner to review and update the script.

Reply:
Hi @0kashi - Can you help with this? Thanks!

Reply:
that link seems to have been deleted.  Though I have updated the script.  Please feel free to test as well.